bp/src/Bluepill.m

```diff
@@ -118,7 +118,7 @@ - (void)beginWithContext:(BPExecutionContext *)context {
 // Retry from the beginning (default) or failed tests only if onlyRetryFailed is true
 - (void)retry {
     // There were test failures. If our failure tolerance is 0, then we're good with that.
-    if (self.failureTolerance == 0) {
+    if (self.failureTolerance <= 0) {
```

Comment:
```
    // There were test failures. Check if it can be retried.
    if (self.retries >= [self.config.errorRetriesCount integerValue] && self.failureTolerance <= 0) {
```
This should fix the underlying bug too. Basically, there should not be a retry if there are no retries left.